crates/sequencer/src/sequencer.rs

```diff
@@ -629,54 +631,194 @@ where
                 .map_err(|e| anyhow!(e))?;
             let (notify, rx) = oneshot_channel();
             let request = BlobWithNotifier { blob, notify };
-            inscription_queue
+            self.da_service
+                .get_send_transaction_queue()
                 .send(request)
                 .map_err(|_| anyhow!("Bitcoin service already stopped!"))?;
-            let tx_id = rx
-                .await
-                .map_err(|_| anyhow!("DA service is dead!"))?
-                .map_err(|_| anyhow!("Send transaction cannot fail"))?;
 
+            info!("Sent commitment to DA queue");
+
+            // TODO: this causes state diff to be lost on restart/error, fix that
+            self.ledger_db.set_state_diff(vec![])?;
+            self.last_state_diff = vec![];
+
+            // Add commitment pending commitments
+            let mut pending_commitments = self.pending_commitments_l2_range.lock().await;
+            pending_commitments.push((*l2_range_to_submit.start(), *l2_range_to_submit.end()));
```

Comment:
maybe push to pending first then delete the state diff stuff 🤔

Reply:
done